js_modules/dagit/packages/core/src/app/__tests__/ExecutionSessionStorage.test.tsx

```diff
@@ -0,0 +1,89 @@
+import {render} from '@testing-library/react';
+import * as React from 'react';
+
+import {TestProvider} from '../../testing/TestProvider';
+import {AppContext} from '../AppContext';
+import {useExecutionSessionStorage} from '../ExecutionSessionStorage';
+
+function mockedLocalStorage() {
+  let store = {};
+  return {
+    setItem(key: string, item: any) {
+      store[key] = item;
+    },
+    getItem(key: string) {
+      return store[key];
+    },
+    removeItem(key: string) {
+      delete store[key];
+    },
+    clear() {
+      store = {};
+    },
+    get length() {
+      return Object.keys(store).length;
+    },
+    key(n: number) {
+      return Object.keys(store)[n];
+    },
+  };
+}
```

Comment:
Looks like you can probably use the real `window.localStorage` instead of a mock after all, since we don't need to spy on it or do other mock stuff with it.